src/internal_rep.rs

```diff
@@ -443,16 +443,16 @@ pub enum AvRuleFlavor {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub struct AvRule<'a> {
```

Comment:
Why removing these references?

Reply:
I think I explained this in the commit message.  Let me know if you're looking for more clarification.

Reply:
I've just discovered that Cow may actually give me exactly what I need here.  Working on an update to that.

Reply:
I've just pushed an update to switch all this to Cows, which seems to work well.  Curious what you think, particularly around putting the Cows in the Context::new() definition.  It's a little annoying from an encapsulation standpoint, but the only place new is used directly outside try_from() is test code, and making the Cow owned vs borrowed decision in try_from() seems natural to me.

Reply:
I agree that it is a bit verbose, but it looks good to me. You can squash this last commit with the previous one though.